core/lib/common/test.js

```diff
@@ -29,19 +29,7 @@ const Context = require('./context');
 
 module.exports = class Test {
 	constructor(id, suite) {
-		this.suite = {
-			rootPath: suite.rootPath,
-			context: suite.context,
-			teardown: {
-				register: (fn, bucket) => {
-					suite.teardown.register(fn, bucket);
-				},
-			},
-			deviceType: suite.deviceType,
-			options: suite.options,
-			state: suite.state,
-		};
-
+		this.suite = suite;
```

Comment:
I would still have liked to see object restructuring (destructuring?) to understand what goes inside the `this.suite` property? I remember we talked about simplifying this since `this.suite` is literally the `Suite` class constructor with extra steps. How about we document the `Suite` class constructor properties instead to make it easy on us when we debug this next time around?

Reply:
@vipulgupta2048  Agreed, I'll document the suite constructor in the next PR and ping you for a review